lib/Sema/CSDiag.cpp

```diff
@@ -1060,19 +1062,38 @@ CalleeCandidateInfo::evaluateCloseness(Type candArgListType,
             // Multiple archetypes, too complicated.
             return { CC_ArgumentMismatch, {}};
           if (rArgType->isEqual(matchingArgType)) {
-            continue;
+            if (nonSubstitutableArgs == 0) {
+              continue;
+            } else {
```

Comment:
No need for the } else {  after a continue.  Please de-indent the ++nonSubstitutableArgs; and fallthrough comment on the next line since that is how flow continues.